ament_index_cpp/test/utest.cpp

```diff
@@ -266,3 +276,50 @@ TEST(AmentIndexCpp, get_packages_with_prefixes) {
   // baz is found in prefix 2 only
   EXPECT_EQ(generate_subfolder_path("prefix2"), packages_with_prefixes["baz"]);
 }
+
+TEST(AmentIndexCpp, has_empty_name){
+  EXPECT_THROW(
+      ament_index_cpp::has_resource("type", ""),
+      std::runtime_error);
+}
+
+TEST(AmentIndexCpp, has_empty_type){
+  EXPECT_THROW(
+      ament_index_cpp::has_resource("", ""),
+      std::runtime_error);
+}
+
+TEST(AmentIndexCpp, has_unknown_resource) {
+  bool success = ament_index_cpp::has_resource("resource_type4", "bar21");
+  EXPECT_FALSE(success);
+}
+
+TEST(AmentIndexCpp, has_resource){
+  std::string res_type = "resource_type1";
+  std::string res_name = "foo";
```

Comment:
nit: avoid this two intermediate variables, just use the literals where needed (which are used only once).